tools/check-wininstaller-translations.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -eu
+
+BASE_DIR=src/res/translation/wininstaller/
+BASE_LANG=en
+INSTALLERLNG=installerlng.nsi
+BASE_LANGSTRINGS=$(grep LangString "${BASE_DIR}/${BASE_LANG}.nsi" | cut -d' ' -f2)
+EXIT=0
+for LANGUAGE_FILE in $(ls -1 src/res/translation/wininstaller/??.nsi | grep -vF "${BASE_LANG}.nsi"); do
+		echo
+		echo "* ${LANGUAGE_FILE}"
+		echo -n "  - Checking language file is included in ${INSTALLERLNG}... "
+		if grep -qF $(tr '/' '\\' <<<"${LANGUAGE_FILE}") "${BASE_DIR}/${INSTALLERLNG}"; then
+				echo "ok"
+		else
+				echo "ERROR"
+				EXIT=1
+		fi
+
+		echo -n "  - Checking language file for empty strings... "
+		if grep -qF '""' "${LANGUAGE_FILE}"; then
+				echo "ERROR, found empty string(s)"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking for comment leftovers from original file... "
+		if head -n1 "${LANGUAGE_FILE}" | grep -qF English; then
+				echo "ERROR, found 'English' in header"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking for wrong macros... "
+		LANG_MACROS="$(grep -oP '\$\{LANG_[^}]+\}' "${LANGUAGE_FILE}")"
+		if grep ENGLISH <<<"$LANG_MACROS"; then
+				echo "ERROR, found LANG_ENGLISH"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking for multiple lang macros... "
+		if [[ $(sort -u <<<"$LANG_MACROS" | wc -l) != 1 ]]; then
+				echo "ERROR, found multiple LANG_ macros"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking if LANG_ macro is in ${INSTALLERLNG}..."
+		if grep -qi '!insertmacro MUI_LANGUAGE "'$(sort -u <<<"${LANG_MACROS}" | sed -rne 's/\$\{LANG_(.*)\}/\1/p')'"' "$BASE_DIR/${INSTALLERLNG}"; then
+				echo "ok"
+		else
+				echo "ERROR, not found"
+				EXIT=1
+		fi
+
+		echo -n "  - Checking for LangStrings diffs... "
+		OTHER_LANGSTRINGS=$(grep LangString "${LANGUAGE_FILE}" | cut -d' ' -f2)
+		diff=$(diff -u <(echo "$BASE_LANGSTRINGS") <(echo "$OTHER_LANGSTRINGS"))
+		[[ ! "$diff" ]] && echo "ok" && continue
+		echo "warning, differences found:"
+		echo "---"
+		echo "$diff"
```

Comment:
How does the formatting from the diff look here (i.e. are all the line breaks still there)?  If it's lost, just rerun the diff.

Reply:
Line breaks are still there due to proper quoting :)

```
* src/res/translation/wininstaller/de.nsi
  - Checking language file is included in installerlng.nsi... ok
  - Checking language file for empty strings... ok
  - Checking for comment leftovers from original file... ok
  - Checking for wrong macros... ok
  - Checking for multiple lang macros... ok
  - Checking if LANG_ macro is in installerlng.nsi...ok
  - Checking for LangStrings diffs... warning, differences found:
--- /dev/fd/63	2021-02-28 10:06:27.394345680 +0100
+++ /dev/fd/62	2021-02-28 10:06:27.397679038 +0100
@@ -2 +2 @@
-INVALID_FOLDER_MSG
+INVALID_FOLDER_MSGx
```